channels/1ds-post-js/src/HttpManager.ts

```diff
@@ -969,17 +978,17 @@ export class HttpManager {
                                 };
 
                                 let isSync = thePayload.isTeardown || thePayload.isSync;
-                                try {
-                                    sendInterface.sendPOST(payload, onComplete, isSync);
-                                    if (_sendListener) {
-                                        // Send the original payload to the listener
-                                        _sendListener(orgPayloadData, payload, isSync, thePayload.isBeacon);
+                                _preparePayload((processedPayload: IPayloadData) => {
+                                    try {
+                                        sendInterface.sendPOST(processedPayload, onComplete, isSync);
+                                        if (_sendListener) {
+                                            // Send the original payload to the listener
+                                            _sendListener(orgPayloadData, processedPayload, isSync, thePayload.isBeacon);
+                                        }
+                                    } catch (ex) {
+                                        _warnToConsole(_logger, "Unexpected exception sending payload. Ex:" + dumpObj(ex));
                                     }
-                                } catch (ex) {
-                                    _warnToConsole(_logger, "Unexpected exception sending payload. Ex:" + dumpObj(ex));
-
-                                    _doOnComplete(onComplete, 0, {});
```

Comment:
The error case is not calling _doOnComplete(), for 1DS this is REALLY important as it will block any further requests because it tracks the number of "active" connections.